pymc3/plots/__init__.py

```diff
@@ -1,9 +1,37 @@
-from .autocorrplot import autocorrplot
-from .compareplot import compareplot
-from .forestplot import forestplot
-from .kdeplot import kdeplot
-from .posteriorplot import plot_posterior, plot_posterior_predictive_glm
-from .traceplot import traceplot
-from .energyplot import energyplot
-from .densityplot import densityplot
-from .pairplot import pairplot
+"""PyMC3 Plotting.
+
+Plots are delegated to the ArviZ library, a general purpose library for
+"exploratory analysis of Bayesian models." See https://arviz-devs.github.io/arviz/
+for details on plots.
+"""
+try:
+    import arviz as az
+except ImportError:  # arviz is optional, throw exception when used
+
+    class _ImportWarner:
+        def __init__(self, attr):
+            self.attr = attr
+
+        def __call__(self, *args, **kwargs):
+            raise ImportError(
+                "ArviZ is not installed. In order to use `{0.attr}`:\npip install arviz".format(self)
+            )
+
+    class _ArviZ:
+        def __getattr__(self, attr):
+            return _ImportWarner(attr)
+
+    az = _ArviZ()
+
+
+autocorrplot = az.plot_autocorr
```

Comment:
Its tempting to reconcile the names, but I suppose we should just leave things be.

Reply:
from running through the first notebook, `varnames --> var_names` is the worst right now. I am a little tempted to add a decorator just for that case:

```python
def map_args(func):
    @functools.wraps(func)
    def wrapped(*args, **kwargs):
        if 'varnames' in kwargs and 'var_names' not in kwargs:
            warnings.warn('Keyword argument `varnames` renamed to `var_names`, and will be removed in pymc3 3.8.0')
            kwargs['var_names'] = kwargs.pop('varnames')
        return func(*args, **kwargs)
    return wrapped
```

Thoughts on that?